### DISCUSSION TITLE
clang-12

### DISCUSSION THREAD

**Discussion by ghost** [OpenIndiana/oi-userland#6751] [General]
Hi,
It would be great to have clang-12 ported to openindiana;
I am currently trying to port it myself so I can probably PR / help

**Comment by ghost** [comment]
I have started looking around and tried to compile clang12 but a problem occurred at link time : 
![image](https://user-images.githubusercontent.com/43730933/118406741-78df4f00-b67d-11eb-81bc-7113dd6c61b1.png)

Not sure what that's about, ABI mismatch probably ? :thinking:
The shared object files exists and those symbols are there (I checked with objdump and grep), so, no idea what's going on

**Reply by alarcher** [reply]
Try to add -lnsl -lsocket explicitly to linker flags.

**Reply by ghost** [reply]
Yep, this did it, thanks ! (never seen that kind of linker error before, now I know)

**Comment by ghost** [comment]
Alright, I've almost got it 
![image](https://user-images.githubusercontent.com/43730933/118412371-ded9cf80-b699-11eb-9022-55276a13a162.png)

**Reply by ghost** [reply]
Linking /opt/gcc/4.4.4/lib/libgcc_s.so.1 to /usr/lib fixed that issue

**Reply by ghost** [reply]
![image](https://user-images.githubusercontent.com/43730933/118415213-6e3aaf00-b6a9-11eb-89f2-20f2741f3c4c.png)
I'm having these while running `gmake REQUIRED_PACKAGES`, tried installing the package corresponding to these shared object files but nothing resolves it

**Reply by alarcher** [reply]
> Linking /opt/gcc/4.4.4/lib/libgcc_s.so.1 to /usr/lib fixed that issue

This won't work, you need to instruct clang to use the right libraries, this can be done by patching to pass -R/path/to/lib to the linker.

**Reply by alarcher** [reply]
> ![image](https://user-images.githubusercontent.com/43730933/118415213-6e3aaf00-b6a9-11eb-89f2-20f2741f3c4c.png)
> I'm having these while running `gmake REQUIRED_PACKAGES`, tried installing the package corresponding to these shared object files but nothing resolves it

You can amend the manifest to bypass the dependency resolution or add an explicit dependency (check in the python modules installing scripts in /usr/bin, they declare the dependency).

**Reply by ghost** [reply]
> > Linking /opt/gcc/4.4.4/lib/libgcc_s.so.1 to /usr/lib fixed that issue
> 
> This won't work, you need to instruct clang to use the right libraries, this can be done by patching to pass -R/path/to/lib to the linker.

Alright

**Reply by alarcher** [reply]
> > > Linking /opt/gcc/4.4.4/lib/libgcc_s.so.1 to /usr/lib fixed that issue
> > 
> > 
> > This won't work, you need to instruct clang to use the right libraries, this can be done by patching to pass -R/path/to/lib to the linker.
> 
> Alright

Did you apply something similar to https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/components/developer/clang-90/patches/09-rpath.patch ?

**Reply by ghost** [reply]
> > > > Linking /opt/gcc/4.4.4/lib/libgcc_s.so.1 to /usr/lib fixed that issue
> > > 
> > > 
> > > This won't work, you need to instruct clang to use the right libraries, this can be done by patching to pass -R/path/to/lib to the linker.
> > 
> > 
> > Alright
> 
> Did you apply something similar to https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/components/developer/clang-90/patches/09-rpath.patch ?

~~No, but I might need, I've tried applying "-L$(GCC_LIBDIR) -Wl,-rpath,$(GCC_LIBDIR)" to the linker flags and I am getting the same error before as I was before symlinking libgcc's shared object;
Same thing seems to happen with -R$(GCC_LIBDIR)
I know what's going on, I am gonna try to patch the code and see how it goes~~

**Reply by ghost** [reply]
Applying the 09-rpatch worked.
